### PR TITLE
Move protos to a separate crate

### PR DESCRIPTION
**Pull request by nrc**
Makes Raft a workspace project

Motivated by https://github.com/pingcap/kvproto/issues/280

PTAL @Hoverbear @BusyJay

### CODE REVIEW

File: proto/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+[features]
+default = []
+gen = []
+
+[build-dependencies]
+protobuf-build = "0.6"
+
+[dependencies]
+bytes = "0.4.11"
+lazy_static = "1.3.0"
```
Comment (BusyJay) [inline proto/Cargo.toml]: Why does it depend on lazy_static?

Comment (nrc) [reply proto/Cargo.toml]: Some of the rust-protobuf API we are emulating requires an initialised default with static lifetime, lazy_static is the best way to support that.

Comment (BusyJay) [reply proto/Cargo.toml]: Interesting. It may also be the reason why it doesn't perform as expected.

Comment (Hoverbear) [reply proto/Cargo.toml]: Could `const fn` help solve it one day?

Comment (nrc) [reply proto/Cargo.toml]: Maybe, I'm not sure. I hope we can just eliminate uses of that wrapper function and then remove lazy static

**Pull request merged**